packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -848,6 +847,46 @@ async function handler(
 
   const allCredentials = await getAllCredentials(organizerUser, eventType);
 
+  let rescheduleUid = reqBody.rescheduleUid;
+  let bookingSeat: Prisma.BookingSeatGetPayload<{ include: { booking: true; attendee: true } }> | null = null;
+  type BookingType = Prisma.PromiseReturnType<typeof getOriginalRescheduledBooking>;
+  let originalRescheduledBooking: BookingType = null;
+
+  if (rescheduleUid) {
+    // rescheduleUid can be bookingUid and bookingSeatUid
+    bookingSeat = await prisma.bookingSeat.findUnique({
+      where: {
+        referenceUid: rescheduleUid,
+      },
+      include: {
+        booking: true,
+        attendee: true,
+      },
+    });
+    if (bookingSeat) {
+      rescheduleUid = bookingSeat.booking.uid;
+    }
+    originalRescheduledBooking = await getOriginalRescheduledBooking(
+      rescheduleUid,
+      !!eventType.seatsPerTimeSlot
+    );
+    if (!originalRescheduledBooking) {
+      throw new HttpError({ statusCode: 404, message: "Could not find original booking" });
+    }
+  }
+
+  const isOrganizerRescheduling = organizerUser.id === userId;
+
+  const attendeeInfoOnReschedule =
```

Comment:
only if the organizer reschedules we want the orignal tz and language. Attendee might be in different tz now and that's why he rescheduled with different tz

Reply:
@emrysal what do you think about that? 

@Udit-takkar's thought: i thought of this edge but if organiser clicks on reschedule from email or just open the reschedule on another device where he is not logged in then we won't be able to know if it is organiser or attendee who is rescheduling